airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/csv/GCSCsvWriter.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.gcs.csv;
+
+import alex.mojaki.s3upload.MultiPartOutputStream;
+import alex.mojaki.s3upload.StreamTransferManager;
+import com.amazonaws.services.s3.AmazonS3;
+import io.airbyte.integrations.destination.gcs.GCSDestinationConfig;
+import io.airbyte.integrations.destination.gcs.GCSFormat;
+import io.airbyte.integrations.destination.gcs.writer.BaseGCSWriter;
+import io.airbyte.integrations.destination.gcs.writer.GCSWriter;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.sql.Timestamp;
+import java.util.UUID;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.QuoteMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GCSCsvWriter extends BaseGCSWriter implements GCSWriter {
```

Comment:
This class is an exact copy of the `S3CsvWriter`. It does not work for GCS. The client passed into the constructor is for S3, and I think `StreamTransferManager` only works for S3 as well.

Reply:
As per [google documentation](https://cloud.google.com/storage/docs/migrating#migration-simple) the client passed into the constructor can be an S3 client as long as the endpoint is GCS (other than HMAC keys of course). Let me test it again (I only tried once or twice) and confirm

Reply:
Wow, this is very handy. Sorry that I did not know this.

Reply:
Not a problem, I found out 2 weeks ago and was amazed! :)